### DISCUSSION TITLE
Add sub-reigons to an already created schematic

### DISCUSSION THREAD

**Discussion by Pow1270** [maruohon/litematica#669] [Ideas]
I noticed that you can't add a sub-reigon to an already created schematic file. If this feature was available it would, for example, be easier to add a differnent colour design to a base farm.

If I want my nether themed design on my farm i will then activate only sub-region "_Farm (base)_" and "_Design (nether themed)_", All other sub-regoins should be hidden. This is how I would do it:

Sub-reigons (3):

- [x] Farm (base)
- [x] Design (nether themed)
- [ ] Design (ocean themed)

But then I realise I want an end themed design, but because I already created the schematic I cannot add another sub-region. So please add a feature so it's possible to add a sub-region to an already existing schematic file so customisation is easier.

**Comment by maruohon** [comment]
I added this to my todo list. So hopefully one day...

**Reply by Pow1270** [reply]
Could you also add to you todo list to change the material list feature so I just see the material list for the loaded sub-regions, not the material for all of them? ;)

**Reply by maruohon** [reply]
By loaded do you mean only in loaded chunks? Because if you toggle of some regions, then I would assume those won't show up, as the placement-based list counts the blocks from the schematic world, and the blocks for disabled regions won't be in the world.

**Reply by Pow1270** [reply]
Huh, that's weird. Maybe just a bug or something but for me I see all the materials required, both for the toggeld sub-regions and the ones i have hidden.

**Reply by Pow1270** [reply]
Well I have now looked at it some more and I don't know anymore. I can for example see that I need 407 spruce planks but 0 black stained glass. In the build I have more glass then planks. And the 3 hidden sub-regions are almost identical to the one I have built

**Reply by maruohon** [reply]
Are you looking at a material list generated directly for a schematic file (opened via the Material List button in the Load Schematics menu), or a Material List generated for a placement (opened initially via the placement configuration menu's Material List button, or using the hotkey when you hadn't opened it via any menu yet since the placement was created or the game started)?

The file-based material list will not care about any toggled sub-regions or render layers etc., it's only generated once directly from the schematic file. Only the placement-based material list will read the blocks from the schematic world, and it will track build progress from the actual client world.

Also do you have the list in All or Render Layers mode?

**Reply by Pow1270** [reply]
I have the list in "All" mode and I use both the hotkey and the placement material list and they give the same result since both open up the placement-based material list as I have understood it.

**Reply by latipun7** [reply]
In the current version (v0.14.3), is there a way to rename Regions and Sub-regions on already created schematics?
It would be extremely helpful to have this along with the ability to edit/add sub-regions on already created schematics.

Thank you for this great mod ✨

**Reply by maruohon** [reply]
No, that's not possible in the current versions. You would have to rename  the regions via an NBT editor such as NBTExplorer.
